Add: Header 및 Footer Component 개발 + Layout 개발

hyeonjin6530 opened the pull request:
Changes to be committed:
       modified:   package-lock.json
       modified:   package.json
       modified:   src/Theme.js -> black 추가

       <아이콘 사용을 위한 svg 파일>
       new file:   src/assets/alram.svg
       new file:   src/assets/back.svg
       new file:   src/assets/home.svg
       new file:   src/assets/list.svg
       new file:   src/assets/mypage.svg
       new file:   src/assets/setting.svg
       new file:   src/assets/upload.svg

       <layout 제작(Header와 Footer)>
       new file:   src/components/layout/Footer.js
       new file:   src/components/layout/Header.js
       new file:   src/components/layout/Layout.js

       <layout 테스트 및 recoil 사용 테스트>
       modified:   src/pages/Main.js

       <recoil을 통한 전역변수관리>
       new file:   src/recoil/recoil.js

       modified:   yarn.lock

VarGun reviewed (commented):
1. layout 수정 : 


 ```
    <div>
        <Header {...headerProps} />
        <main>{children}</main>
        <Footer isAdmin={isAdmin} />
    </div>
```
 


위처럼 수정해야 내부에 다른 애들을 띄울 수 있어요.

2. isAdmin -> recoil 로 관리하도록 : 사용 도중에 관리자모드를 바꿀 수 있어야 하므로 

3. Header : border-bottom 추가

4. selectedButtonState - default : 'home'

5. App.js 수정 : 

```
   <ThemeProvider theme={theme}>
      <RecoilRoot>
        <div className="App">
        </div>
      </RecoilRoot>
    </ThemeProvider>
```

RecoilRoot 는 최상위 컴포넌트에 위치해야 합니다.

6. 따라서 Main.js 에서 RecoilRoot 삭제

VarGun reviewed (commented):
### Footer

- StyledFooter
    - 이름 변경 Slack 中 “Styled 어쩌고도 금지”
    - 내부 요소와의 간격이므로 Button 들을 다 감싸는 div 를 만들고 StyledFooter 에 padding 을 주는 방식으로 구현.
    - Button 들을 감싸는 div 의 gap 으로 Button 들끼리 간격 조정
- Button
    - margin 삭제
- isAdminState
    - default 가 false 인데도 관리자 모드 footer 가 뜹니다.
![image](https://github.com/2023-2-WINK-Project-Team1/front-end/assets/88530565/f928656e-404c-46ff-be3f-bfa703a46b64)
![image](https://github.com/2023-2-WINK-Project-Team1/front-end/assets/88530565/eaf85d5d-730a-48ae-928d-4d0631a3ea6e)




### Header

- 마찬가지로 이름 변경
- 마찬가지로 내부 요소 배치 관련해서 Footer 와 같이 수정
    - margin 들 삭제

### 주석 Jonna Great

VarGun reviewed (commented):
### Header.js

HeaderContainer

- justify-content : 속성 중복

ButtonContainer

- gap 대신 justify-content : space-between
- width: 100%, padding: 0 16px; 로 간격 조정

### Footer.js

isAdmin 을 props 로 넘겨줄 필요 없음

- isAdminState 가 바뀔 때마다 알아서 리렌더링 됨.

ButtonContainer

- 마찬가지로 gap 대신 justify-content : space-between
- width: 100%, padding: 0 16px; 로 간격 조정

hyeonjin6530 committed:
Fix: Header 및 Footer Component 개발 + Layout 개발
<props로 전달되는 isAdmin 제거 및 css 수정>
       modified:   src/components/layout/Footer.js
       modified:   src/components/layout/Layout.js

<css 수정>
       modified:   src/components/layout/Header.js

<수정사항 테스트>
       modified:   src/pages/Main.js

VarGun committed:
Merge branch 'develop' into front-end-25
